app/javascript/miq-redux/middleware.js

```diff
@@ -5,7 +5,12 @@ import promiseMiddleware from 'redux-promise-middleware';
 export const taggingMiddleware = store => next => action => {
   const { type, meta, tagCategory, tag } = action;
   if (meta && meta.url) {
-    const params = {id: store.getState().tagging.appState.affectedItems[0], cat: tag.tagCategory.id, val: tag.tagValue.id, check: 1, tree_typ: 'tags' };
+    const ids = [...store.getState().tagging.appState.assignedTags.map( t => t.values.map(val => val.id)), tag.tagValue.id].flat();
+    console.log(ids);
+    const params = (meta.type === 'provision'
```

Comment:
This is a 🦛 step backwards 🦛 with the ternary operator... :red_circle: :black_flag: @himdel @martinpovolny @Hyperkid123 any idea how else we could do this?

Reply:
What do you mean by step back? There not much options with expression like this. You can do classic `if` statement. Or `switch` or maybe invoke a function like this:
```javascript
// or extract it somewhere as a helper function
params = ((type, state) => ({
  provision: state.something...,
  default: 'b'
})[type || 'default'])(meta.type, state)
```

Reply:
My problem here is the special case when the tagging is on the provisioning screen. Anything like this has the potential to grow into a huge case with a ton of exceptions

Reply:
Ah ok, i would suggest then to add a Prop to components that are using this and create the params inside the component and send it as a (meta)parameter to the action.

Reply:
There should be a convention for this.

Every form not just talking to the API will have the same problem - the definiton of what to send via http can not be part of the component itself.

Passing a function to transform state to request data sounds reasonable, but maybe we should really do it the same way in all the react forms.